### PR TITLE
Handle more than one person per citation

### PR DESCRIPTION
**Pull request by ctwoolsey**

### CODE REVIEW

**Comment by ctwoolsey** [comment]
This PR depends on API PR#54  and also should be merged after PR#124

File: app/views/citationInfo.html [hunk 1]
```diff
@@ -3,6 +3,7 @@
 </div>
 <div class="sub-page-container">
   <div class="ticket-results-container container-fluid topMargin alignContainerToPage">
+    <div class="well well-sm warning-red-text" ng-if="ctrl.issueMultiplePeopleWarning()">Multiple people may have been returned for the tickets.  Please make sure you check that you are looking at the correct ticket.</div>
```
Comment (abobwhite) [inline app/views/citationInfo.html]: This wording is a little less than ideal (doesn't read clearly). Maybe "The results may contain tickets for multiple people. Please make sure you look closely for the correct ticket"?

File: app/scripts/controllers/citationInfo.ctrl.js [hunk 2]
```diff
@@ -7,6 +7,24 @@ angular.module('yourStlCourts').controller('CitationInfoCtrl', function (faqData
   ctrl.paymentUrl = "";
   ctrl.citationCourtLocations = {};
 
+  ctrl.issueMultiplePeopleWarning = function(){
```
Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: So....we don't want to just show groupings of tickets by people? If we always show groupings, then it handles every case and we don't have to show a message. It's implicitly indicated separately.

File: app/views/citationInfo.html (hunk 1 above)
Comment (abobwhite) [inline app/views/citationInfo.html]: That is, of course, if we keep this message.

File: app/scripts/controllers/citationInfo.ctrl.js [hunk 3]
```diff
@@ -7,6 +7,38 @@ angular.module('yourStlCourts').controller('CitationInfoCtrl', function (faqData
   ctrl.paymentUrl = "";
   ctrl.citationCourtLocations = {};
 
+  ctrl.groupCitationsByDL = function(){
+    var dlNum = "";
+    var dlState = "";
+    var groupedCitations = new Array();
+
+    for(var citationIndex = 0; citationIndex < citations.length; citationIndex++){
+      var groupedCitation = citations[citationIndex];
```
Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: In general, for loops aren't really used anymore unless the index is needed. I see in this case you're not using the index. Instead, just use `citations.forEach(function(citation) { ... });`

File: app/scripts/controllers/citationInfo.ctrl.js [hunk 4]
```diff
@@ -7,6 +7,38 @@ angular.module('yourStlCourts').controller('CitationInfoCtrl', function (faqData
   ctrl.paymentUrl = "";
   ctrl.citationCourtLocations = {};
 
+  ctrl.groupCitationsByDL = function(){
+    var dlNum = "";
+    var dlState = "";
+    var groupedCitations = new Array();
```
Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: This is really uncommon. In JavaScript, just use `[]` instead of `new Array()`

Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: https://coderwall.com/p/h4xm0w/why-never-use-new-array-in-javascript

File: app/scripts/controllers/citationInfo.ctrl.js [hunk 5]
```diff
@@ -7,6 +7,38 @@ angular.module('yourStlCourts').controller('CitationInfoCtrl', function (faqData
...
+
+    for(var citationIndex = 0; citationIndex < citations.length; citationIndex++){
+      var groupedCitation = citations[citationIndex];
+      dlNum = groupedCitation.drivers_license_number;
+      dlState = groupedCitation.drivers_license_state;
+      if (dlNum === ""){
+        //in the event the defendant does not have a DL Num assign one so the array has an index
+        dlNum = "A1";
```
Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: Why `A1` ? Is that picked at random? If we're using something as a placeholder, it should be clearer that it's a placeholder. And you'll want to define it in const pattern like `PLACEHOLDER_DL_NUM = 'NO_DL_NUM'` or something

File: app/scripts/controllers/citationInfo.ctrl.js [hunk 6]
```diff
@@ -7,6 +7,38 @@ angular.module('yourStlCourts').controller('CitationInfoCtrl', function (faqData
...
+      }
+      if (!groupedCitations[dlNum+dlState]){
+        groupedCitations[dlNum+dlState] = new Array();
```
Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: Again here `[]`

File: app/scripts/controllers/citationInfo.ctrl.js [hunk 7]
```diff
@@ -7,6 +7,38 @@ angular.module('yourStlCourts').controller('CitationInfoCtrl', function (faqData
...
+      }
+      groupedCitations[dlNum+dlState].push(groupedCitation);
+    }
+
+    //re-index array so it uses ints as index
+    var indexGroupedCitations = new Array();
+    for(var i in groupedCitations){
```
Comment (abobwhite) [inline app/scripts/controllers/citationInfo.ctrl.js]: Again here with `[]` and the `for` loop

File: app/views/citationInfo.html [hunk 8]
```diff
@@ -3,36 +3,42 @@
...
-              </span>
-              <span class="noViolations" ng-if="!ctrl.hasViolations(citation)">
-                Violation data unavailable. Please contact your court.
-              </span>
-            </td>
-          </tr>
-          </tbody>
-        </table>
+        <span ng-repeat="citationGrouping in ctrl.groupedCitations">
+          <h3 class="no-indent" ng-if="ctrl.groupedCitations.length > 1">
```
Comment (abobwhite) [inline app/views/citationInfo.html]: Why are we only showing the name if there is more than 1 grouped citations? I would think we'd show the name in all cases. What's the harm?

Comment (ctwoolsey) [inline app/views/citationInfo.html]: The name is already part of the citation and it appears cleaner when there is no explicit need to warn people that these tickets might not be for them.

**Pull request merged**